Minutes 2022-04-06 (#294)
solid/authorization-panel: meetings/2022-04-06.md

* SC: ACP/WAC/ODRL might be required for different bits and if it's clear how they can work together, then we'll go with that.
* MB: I think it's pretty clear, elf, anything else?
* eP: I think I'll still create an issue and PR, I prefer having push back on that PR.
* SC: I'm telling you that solving [121](https://github.com/solid/authorization-panel/issues/121) is a good way to go at it. This is not just about WAC or ACP... Because we need a proper view of extensibility.
* SC: I'll have to jump off.
* eP: It would help to document ACP in ESS.
* MB: Everything supports either or, i don't know of a pod that have WAC and ACP at the same time.
* WT: Would there be a benefit of using both WAC and ACP in the same solid storage?
* MB: There is a lot in WAC specification that is not about expressing permissions: effective ACL discovery, access modes discovery (wac-allow header), access modes used... The declarative permissions bit is quite small in WAC and that is most of ACP (it's a domain model).
* ...: It's probably good to choose between one domain model for declaring permissions or the other (ACP or ACL).

> * eP: I also created https://github.com/solid/specification/issues/377 talking about storage types.

Reviewer: ```suggestion
* eP: I also created [issue #377](https://github.com/solid/specification/issues/377) talking about storage types.
```

> * WT: I agree that after 0.9 can be 0.10 0.11 etc. It's not penultimate to 1.0.

Reviewer: ```suggestion
* WT: I agree that after 0.9 can be 0.10, 0.11, etc. It's not penultimate to 1.0.
```

* eP: Matthieu as editor of ACP do you see it ready for FPWD? We can think of it as 0.1

> * MB: I don't know what the implication is of tagging it this way. I find answering #121 more important.

Reviewer: ```suggestion
* MB: I don't know what the implication is of tagging it this way. I find answering [issue #121](https://github.com/solid/authorization-panel/issues/121) more important.
```

### ACP Context attributes

https://solid.github.io/authorization-panel/acp-specification/#context-attributes

https://github.com/solid/specification/issues/379#issuecomment-1090235543

* eP: I'd like to discuss it considering that AS and not RS might be evaluating the policies.

> * eP: My understanding is: the RS provides context to the AS and AS runs the ACP engine and respond to the resource server with allowed or disallowed?

Reviewer: ```suggestion
* eP: My understanding is: the RS provides context to the AS, and AS runs the ACP engine and responds to the resource server with allowed or disallowed?
```

* MB: The interface between AS and RS in the response is mainly the set of access modes. RS provides description of the request (the context), including full representation of the effective ACR. The AS reponds with set of access modes, which may be empty.
* eP: context includes `acp:mode` why AS would repond with a list of modes. This is good reason to say we work on ???.

> * MB: The interface between AS and RS would be describing the request and getting back set of access modes. Especially even more in the response. Both should be understood by the AS and RS.

Reviewer: ```suggestion
* MB: The interface between AS and RS would be describing the request and getting back set of access modes. Especially in the response. Both should be understood by the AS and RS.
```

> * eP: I try to understand the case. Say on a get request, the server creates a context and post the access mode (Read), the response to get could include that you can do more than Read.

Reviewer: ```suggestion
* eP: I try to understand the case. Say on a `GET` request, the server creates a context and posts the access mode (`Read`); the response to `GET` could include that you can do more than `Read`.
```

> * MB: So my intent in describing (potentially) the access modes requested is that it might enable some interesting optimisations on the AS side. Once you found answers you are done. But if you didn't specify any mode, you might get all the possible ones back.

Reviewer: ```suggestion
* MB: So my intent in (potentially) describing the access mode(s) requested is that it might enable some interesting optimizations on the AS side. Once you've found answers, you are done. But if you didn't specify any mode, you might get all the possible ones back.
```

* eP: Is the AS responsible to fetch all the Policies the ACR depends on?

> * MB: RS and AS could have a dialog and AS can query RS multiple types. RS may entrust AS with entirely with the authz graph. The important part is that there is a trust relation between RS and AS. My preference would be to leave the AuthZ graph to the RS since it makes it more portable. 8

Reviewer: ```suggestion
* MB: RS and AS could have a dialog, and AS can query RS multiple types. RS may entrust AS with the entire authz graph. The important part is that there is a trust relation between RS and AS. My preference would be to leave the AuthZ graph to the RS, since that makes it more portable.
```

> * eP: The RS as I understand is not responsible over the policies (say if they are external). Possibly, the AS may not even have access to those resources.

Reviewer: ```suggestion
* eP: The RS, as I understand, is not responsible over the policies (say if they are external). Possibly, the AS may not even have access to those resources.
```

> * MB: I have no strong opinion to any of this. I care that the ACP engine is able to have access to a full authoritative authorization graph (the full ACL representation in other words), and this requires full trust. You can't deference polices comming from about anywhere because it's unsafe.

Reviewer: ```suggestion
* MB: I have no strong opinion to any of this. I care that the ACP engine is able to have access to a full authoritative authorization graph (the full ACL representation in other words), and this requires full trust. You can't dereference polices coming from anywhere because it's unsafe.
```

> * eP: Do you remember that we talked about Data Grants spanning accross multiple Resource Servers? If they define different ASs, what trust is required?

Reviewer: ```suggestion
* eP: Do you remember that we talked about Data Grants spanning across multiple Resource Servers? If they define different ASs, what trust is required?
```
